pilot/pkg/xds/filters/filters.go

```diff
@@ -264,6 +264,42 @@ var (
 				}),
 		},
 	}
+
+	GatewayMtlsMetadataFilter = &hcm.HttpFilter{
+		Name: MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
+				map[string]any{
+					"upstream_discovery": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+						map[string]any{
+							"workload_discovery": map[string]any{},
+						},
+					},
+					"upstream_propagation": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
```

Comment:
I am concerned this is inconsistent with `SkipIstioMXHeaders`

Reply:
agreed, curious if call to `httpbin.org/headers` through egress gateway end up receiving the `X-Envoy-Peer-Metadata` or `X-Envoy-Peer-Metadata-Id`

Reply:
This PR doesn't change the Upstream behaviour for the metadata filter. The metadata headers shouldn't leak to external services because of the current change.

Reply:
The filter will make gateway to do metadata exchange with upstream, which could be external service

Reply:
what does the sidecar do now?  @kyessenov

Reply:
> The filter will make gateway to do metadata exchange with upstream, which could be external service

Would the MX filter inject headers upstream because the downstream peer will send the metadata headers? Can we configure different configurations for Egress and Ingress gateways if that is the case? Ideally, Egress Gateway shouldn't have `upstream_propagation` configuration.

Reply:
Yes, that's my concern. We should document the behavior for different cases.

Ingress gateway: donot need dowmstream mx

Egress gateway: donot need upstream mx